.github/workflows/test_branches.yml

```diff
@@ -52,32 +52,32 @@ jobs:
           PACKAGES: glpk
 
         - os: ubuntu-latest
-          python: 3.8
-          other: /mpi
-          mpi: 3
+          python: 3.9
+          other: /conda
           skip_doctest: 1
           TARGET: linux
           PYENV: conda
           PACKAGES: mpi4py openmpi
 
         - os: ubuntu-latest
-          python: 3.7
-          other: /conda
+          python: 3.8
+          other: /mpi
+          mpi: 3
           skip_doctest: 1
           TARGET: linux
           PYENV: conda
           PACKAGES: mpi4py openmpi
 
         - os: ubuntu-latest
-          python: 3.9
+          python: 3.7
```

Comment:
Any particular reason for changing the Python version here or is this a weirdness in GitHub's diff? (Same question on line 80 for the ubuntu-latest cython job)

Reply:
I was synchronizing the two test files, and trying to think through a good distribution of tests to python versions (especially since we are about to drop 3.6 and are adding 3.10)